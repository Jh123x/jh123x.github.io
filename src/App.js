import { useState, useEffect } from "react";
import Sidebar from "./Components/Sidebar";
import styled from "styled-components";
import { Brightness4, Menu as MenuIcon } from "@mui/icons-material";
import { Switch, IconButton } from "@mui/material";
import HomePage from "./Pages/HomePage";
import AboutPage from "./Pages/AboutPage";
import PortfoliosPage from "./Pages/PortfoliosPage";
import ResumePage from "./Pages/ResumePage";
import { Routes, Route } from "react-router-dom";

const App = () => {
  const [checked, setChecked] = useState(false);
  const [navToggle, setNavToggle] = useState(false);

  useEffect(() => {
    document.documentElement.className = checked ? "light-theme" : "dark-theme";
  }, [checked]);

  return (
    <div className="App">
      <Sidebar navToggle={navToggle} />

      <div className="theme">
        <div className="light-dark-mode">
          <div className="left-content">
            <Brightness4 />
          </div>
          <div className="right-content">
            <Switch
              value=""
              checked={checked}
              inputProps={{ "aria-label": "" }}
              size="medium"
              onClick={() => setChecked(!checked)}
            />
          </div>
        </div>
      </div>

      <div className="ham-burger-menu">
        <IconButton onClick={() => setNavToggle(!navToggle)}>
          <MenuIcon />
        </IconButton>
      </div>

      <MainContentStyled>
        <Routes>
          <Route path="/">
            <Route index element={<HomePage />} />
            <Route path="resume" element={<ResumePage />} />
            <Route path="about" element={<AboutPage />} />
            <Route path="portfolios" element={<PortfoliosPage />} />
          </Route>
        </Routes>
      </MainContentStyled>
    </div>
  );
};

const MainContentStyled = styled.main`
  position: relative;
  margin-left: 16.3rem;
  min-height: 100vh;
  @media screen and (max-width: 1200px) {
    margin-left: 0;
  }
  .lines {
    position: absolute;
    min-height: 100%;
    width: 100%;
    display: flex;
    justify-content: space-evenly;
    opacity: 0.4;
    z-index: -1;
    .line-1,
    .line-2,
    .line-3,
    .line-4 {
      width: 1px;
      min-height: 100vh;
      background-color: var(--border-color);
    }
  }
`;

export default App;
