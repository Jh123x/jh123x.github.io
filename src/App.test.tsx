import * as React from "react";
import App from "./App";
import { describe, expect, test } from "@jest/globals";

describe("Render App", () => {
  test("renders App component", () => {
    const app = <App />;
    expect(app).toMatchSnapshot();
  });
});
