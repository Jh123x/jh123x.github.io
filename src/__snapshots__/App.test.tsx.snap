// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App renders App component 1`] = `<App />`;
